modules/ROOT/pages/dataweave-type-reuse-xml-schema.adoc

```diff
@@ -0,0 +1,429 @@
+= Reusing an XML Schema
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+You can import XML schema files (`.xsd`) inside your DataWeave script as modules by using the `xmlschema!` module loader. This loader enables you to use types that are declared on your schema inside DataWeave directly.
+DataWeave loads your XSD file and translates declarations inside your file into DataWeave type directives that can be accessed in the same way as types from any other DataWeave module. Use them to build new types, type check your variables, do pattern matching, or declare new functions that use types. There are no restrictions on how to use these types.
+
+The following example shows how to look for an XSD file in the path `org/weave/myfolder/User.xsd` being the first path the root of the classpath of your application:
+
+----
+import * from xmlschema!org::weave::myfolder::User
+----
+
+== Import Syntax
+
+To import the XML schema types, as shown above, use the following syntax, where:
+
+* `typeToImport`: You can use `\*` to import all types defined in the schema, or to export a single type from the schema, for example, `Root`. You can also import XML schema types with a different name, for example, `Root as Person`. This way  you can reference the type with that name in the script.
+* `pathToXsdSchemaFile`: To specify the path to the schema file, replace the file separators with `::` and remove the `.xsd` extension from the file name. For example, if the path to the schema is `example/schema/Person.xsd`, use `example::schema::Person`.
+
+----
+import _typesToImport_ from xmlschema!_pathToXsdSchemaFile_
+----
+
+The following example shows how to import a type:
+
+----
+import * from xmlschema!example::schema::Person
+----
+
+== Use Your Types in a DataWeave Script
+
+This example shows how to use types to import XML schema files into your DataWeave script.
+
+.XSD File (`User.xsd`):
+[source,XML]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified"
+           xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <xs:element name="Root">
+        <xs:complexType>
+            <xs:choice>
+                <xs:element name="employee" type="employee"/>
+                <xs:element name="member" type="member"/>
+            </xs:choice>
+            <xs:attribute name="id" use="required"/>
+        </xs:complexType>
+    </xs:element>
+    <xs:complexType name="employee">
+        <xs:sequence>
+            <xs:element name="name" type="xs:string"/>
+        </xs:sequence>
+    </xs:complexType>
+
+
+    <xs:complexType name="member">
+        <xs:sequence>
+            <xs:element name="id" type="xs:string"/>
+        </xs:sequence>
+    </xs:complexType>
+</xs:schema>
+----
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from xmlschema!org::mule::weave::v2::module::xmlschema::engine::pattern_matching::User
```

Comment:
Where does User come from in this? xmlschema!org::mule::weave::v2::module::xmlschema::engine::pattern_matching::User

I don't see it in the schema, for example. I see Root as the element name.
Does this example point to a path in our source code? Should we use that path?
@martincousido

Reply:
That path is the location of the schema file, it doesn't appear inside it.
It is a test that he have in our repo. I think we could change it to:
```suggestion
import * from xmlschema!org::weave::myfolder::User
```
That is the same as the first example at the beginning of this page